class.jetpack-geo-locate.php

```diff
@@ -0,0 +1,409 @@
+			return $content;
+		}
+
+		$microformat = sprintf(
+			'<div id="geo-post-%d" class="geo geo-post" style="display: none">',
+			esc_attr( $this->get_post_id() )
+		);
+
+		$microformat .= sprintf(
+			'<span class="latitude">%s</span>',
+			esc_html( $meta_values['latitude'] )
+		);
+
+		$microformat .= sprintf(
+			'<span class="longitude">%s</span>',
+			esc_html( $meta_values['longitude'] )
+		);
+
+		$microformat .= '</div>';
+
+		return $content . $microformat;
+	}
+
+	/**
+	 * Register a range of hooks for integrating geo data with various feeds.
+	 */
+	public function register_rss_hooks() {
+		add_action( 'rss2_ns', array( $this, 'rss_namespace' ) );
+		add_action( 'atom_ns', array( $this, 'rss_namespace' ) );
+		add_action( 'rdf_ns', array( $this, 'rss_namespace' ) );
+		add_action( 'rss_item', array( $this, 'rss_item' ) );
+		add_action( 'rss2_item', array( $this, 'rss_item' ) );
+		add_action( 'atom_entry', array( $this, 'rss_item' ) );
+		add_action( 'rdf_item', array( $this, 'rss_item' ) );
+	}
+
+	/**
+	 * Add the georss namespace during RSS generation.
+	 */
+	public function rss_namespace() {
+		echo 'xmlns:georss="http://www.georss.org/georss" xmlns:geo="http://www.w3.org/2003/01/geo/wgs84_pos#" ';
+	}
+
+	/**
+	 * Output georss data for RSS items, assuming we have data for the currently rendered post and
+	 * that data as marked as public.
+	 */
+	public function rss_item() {
+		$meta_values = $this->get_meta_values( $this->get_post_id() );
+
+		if ( ! $meta_values['is_public'] ) {
+			return;
+		}
+
+		printf(
+			"\t<georss:point>%s %s</georss:point>\n",
+			ent2ncr( esc_html( $meta_values['latitude'] ) ),
+			ent2ncr( esc_html( $meta_values['longitude'] ) )
+		);
+
+		printf( "\t\t<geo:lat>%s</geo:lat>\n", ent2ncr( esc_html( $meta_values['latitude'] ) ) );
+		printf( "\t\t<geo:long>%s</geo:long>\n", ent2ncr( esc_html( $meta_values['longitude'] ) ) );
+	}
+
+	/**
+	 * Enqueue CSS for rendering post flair with geo-location.
+	 */
+	public function enqueue_scripts() {
+		wp_enqueue_style( 'dashicons' );
+	}
+
+	/**
+	 * If we're rendering a single post and public geo-location data is available for it,
+	 * include the human-friendly location label in the output.
+	 *
+	 * @param string $html
+	 *
+	 * @return string
+	 */
+	public function the_content_location_display( $content ) {
+		if (! $this->is_single() ) {
+			return $content;
+		}
+
+		$meta_values = $this->get_meta_values( $this->get_post_id() );
+
+		if ( ! $meta_values['is_public'] ) {
+			return $content;
+		}
+
+		// If the location has not been labeled, do not show the location
+		if ( ! $meta_values['label'] ) {
+			return $content;
+		}
+
+		$html = '<div class="post-geo-location-label geo-chip">';
+		$html .= '<span class="dashicons dashicons-location" style="vertical-align: text-top;"></span> ';
+		$html .= esc_html( $meta_values['label'] );
+		$html .= '</div>';
+
+		return $content . apply_filters( 'geo_location_display', $html, $meta_values );
```

Comment:
Could you add a docblock before the filter, so our parser can pick this up and generate a codex page? Here is an example:
https://github.com/Automattic/jetpack/blob/6.1.1/functions.opengraph.php#L206
https://developer.jetpack.com/hooks/jetpack_open_graph_tags/

Reply:
Absolutely.  Thank you for pointing out that example.  This docblock is added in 
0c21d03.  I left out the `@since` annotation because I'm not sure what version/release we're targeting for this feature's inclusion.  If you can think of a sane value for it, let me know.